From 29d683d07610b895a57f61c54c6594f945a9cd91 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 28 Aug 2014 03:04:05 -0400
Subject: [acpi] Create symlinks in acpi_bind_one() under physical_node_lock

Message-id: <20140828030405.2117.19442.stgit@gir.stowe>
Patchwork-id: 89293
O-Subject: [RHEL7.1 PATCH 05/30] ACPI: Create symlinks in acpi_bind_one() under physical_node_lock
Bugzilla: 1128632
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1128632
Upstream Status: 4005520648c7d6cf28e74addb52bc4a793eea3eb

commit 4005520648c7d6cf28e74addb52bc4a793eea3eb
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Wed Aug 7 01:19:37 2013 +0200

    ACPI: Create symlinks in acpi_bind_one() under physical_node_lock

    Put the creation of symlinks in acpi_bind_one() under the
    physical_node_lock mutex of the given ACPI device object, because
    that is part of the binding operation logically (those links are
    already removed under that mutex too).

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
    Acked-by: Toshi Kani <toshi.kani@hp.com>
    Acked-by: Yasuaki Ishimatsu <isimatu.yasuaki@jp.fujitsu.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/glue.c b/drivers/acpi/glue.c
index 92cacb1..914a346 100644
--- a/drivers/acpi/glue.c
+++ b/drivers/acpi/glue.c
@@ -242,8 +242,6 @@ int acpi_bind_one(struct device *dev, acpi_handle handle)
  list_add(&physical_node->node, physnode_list);
  acpi_dev->physical_node_count++;
 
- mutex_unlock(&acpi_dev->physical_node_lock);
-
  if (!ACPI_HANDLE(dev))
   ACPI_HANDLE_SET(dev, acpi_dev->handle);
 
@@ -253,6 +251,8 @@ int acpi_bind_one(struct device *dev, acpi_handle handle)
  retval = sysfs_create_link(&dev->kobj, &acpi_dev->dev.kobj,
   "firmware_node");
 
+ mutex_unlock(&acpi_dev->physical_node_lock);
+
  if (acpi_dev->wakeup.flags.valid)
   device_set_wakeup_capable(dev, true);
 
-- 
1.7.1